From b9a5b646b270b6772f1340e5ef7ebac10ae0420b Mon Sep 17 00:00:00 2001
From: Mahesh Reddy Kodidi <mahreddy@qti.qualcomm.com>
Date: Wed, 28 May 2025 15:33:53 +0530
Subject: [PATCH] Revert "soc: qcom: memory_dump: fix register dump table entry
 failed"

This reverts commit bef646eeb61038dcf048d39deba76ecbefc4faff.

Reason for revert: <It is causing regression with tzpu_ctx0 with dump-id = 0x0>

Change-Id: I66baa2353fc527928d5a1f3c0a150a92994cd444
Signed-off-by: Mahesh Reddy Kodidi <mahreddy@qti.qualcomm.com>
---
 drivers/soc/qcom/memory_dump_v2.c | 23 +++--------------------
 1 file changed, 3 insertions(+), 20 deletions(-)

diff --git a/drivers/soc/qcom/memory_dump_v2.c b/drivers/soc/qcom/memory_dump_v2.c
index 0515ebe3de3d..985bd12e2c06 100644
--- a/drivers/soc/qcom/memory_dump_v2.c
+++ b/drivers/soc/qcom/memory_dump_v2.c
@@ -628,21 +628,8 @@ static int msm_dump_data_add_minidump(struct msm_dump_entry *entry)
 	return msm_minidump_add_region(&md_entry);
 }
 
-static struct msm_dump_entry *msm_dump_entry_get(struct msm_dump_table *table,
-						 struct msm_dump_entry *entry)
-{
-	int i;
-
-	for (i = 0; i < MAX_NUM_ENTRIES; i++) {
-		if (table->entries[i].id == entry->id)
-			return &table->entries[i];
-	}
-
-	return NULL;
-}
-
 static int register_dump_table_entry(enum msm_dump_table_ids id,
-				     struct msm_dump_entry *entry)
+					struct msm_dump_entry *entry)
 {
 	struct msm_dump_entry *e;
 	struct msm_dump_table *table;
@@ -654,15 +641,11 @@ static int register_dump_table_entry(enum msm_dump_table_ids id,
 	if (!table || table->num_entries >= MAX_NUM_ENTRIES)
 		return -EINVAL;
 
-	e = msm_dump_entry_get(table, entry);
-	if (!e) {
-		e = &table->entries[table->num_entries];
-		table->num_entries++;
-	}
-
+	e = &table->entries[table->num_entries];
 	e->id = entry->id;
 	e->type = MSM_DUMP_TYPE_DATA;
 	e->addr = entry->addr;
+	table->num_entries++;
 
 	return 0;
 }
-- 
2.17.1

